toolkit/tools/pkg/imagecustomizerlib/versionsOfToolDependencies.go

```diff
@@ -0,0 +1,95 @@
+package imagecustomizerlib
+
+import (
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+)
+
+func getVersionsOfToolDeps() {
+	// Map of version flags with corresponding packages
+	versionFlags := map[string][]string{
+		"--version": {
+			"qemu-img", "rpm", "dd", "lsblk", "losetup", "sfdisk", "udevadm",
+			"flock", "blkid", "sed", "createrepo", "genisoimage", "parted", "mkfs",
+			"fsck", "fatlabel", "zstd", "veritysetup", "grub-install",
+		},
+		"-version": {
+			"mksquashfs",
+		},
+		"version": {
+			"openssl",
+		},
+		"-V": {
+			"mkfs.ext4", "mkfs.xfs", "e2fsck", "xfs_repair", "xfs_admin",
+		},
+		"none": {
+			"mkfs.vfat", "resize2fs", "tune2fs",
+		},
+	}
+
+	// Get distro and version
+	distro, version := getDistroAndVersion()
+	logger.Log.Debugf("Distro: %s, Version: %s\n", distro, version)
+
+	// Get versions of packages
+	logger.Log.Debugf("Tool Dependencies:")
+	for versionFlag, pkgList := range versionFlags {
+		for _, pkg := range pkgList {
+			version, err := getPackageVersion(pkg, versionFlag)
+			if err != nil {
+				logger.Log.Debugf("%s: not installed or error retrieving version\n", pkg)
```

Comment:
Don't add tail newlines to log messages.

Similar comment elsewhere.

Reply:
Fixed!